docs/source/api-life-cycle.md

```diff
@@ -0,0 +1,148 @@
+4. Once the deprecation period has passed, the API may be removed (See “Deprecation Period” below). Be sure to also remove references from the documentation.
+
+
+We also use deprecation as a way to make breaking changes to an existing interface: for example, if adding a non-optional parameter to a method. To do this without breaking existing users:
+1. In a single commit:
+   - Create a new API that meets the new requirements.
+   - Deprecate the old API and recommend that users move to the new API.
+2. Migrate use cases from the old API to the new API.
+3. Delete the old API after the deprecation period.
+
+
+## How to Mark API State
+
+When possible, the ExecuTorch code uses language-standard ways to annotate API lifecycle state in the code. This makes it easier for IDEs and other tools to communicate state to developers.
+
+<table>
+  <tr>
+   <td><strong>API</strong>
+   </td>
+   <td><strong>Code</strong>
+   </td>
+   <td><strong>Documentation</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Python
+   </td>
+   <td>
+
+<em>Deprecated</em>: use the <a href="https://typing-extensions.readthedocs.io/en/latest/#typing_extensions.deprecated">typing_extensions.deprecated</a> decorator
+
+<em>Experimental</em>: use ExecuTorch's native experimental (TODO not yet implemented) decorator
+
+   </td>
+   <td>
+
+Use <code>.. warning::</code> in the docstrings of deprecated and experimental APIs. See <a href="https://github.com/pytorch/pytorch/blob/cd8bbdc71a0258292381a7d54c8b353988d02ff4/torch/nn/utils/stateless.py#L170">example usage</a>.
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td>C++
+   </td>
+   <td>
+
+Use the <code>__ET_DEPRECATED</code> or (TODO not yet implemented) <code>__ET_EXPERIMENTAL</code> macros. See <a href="https://github.com/pytorch/executorch/blob/8e0f856ee269b319ac4195509cf31e3f548aa0e8/runtime/executor/program.h#L81">example usage</a>.
+</ul>
+   </td>
+   <td>
+
+Start Doxygen comments with "DEPRECATED:" or "EXPERIMENTAL:". See <a href="https://github.com/pytorch/executorch/blob/9d859653ae916d0a72f6b2b5c5925bed38832140/runtime/executor/program.h#L139">example usage</a>.
+
+   </td>
+  </tr>
+  <tr>
+   <td>Android
+   </td>
+   <td>Deprecated: use <a href="https://docs.oracle.com/javase/9/docs/api/java/lang/Deprecated.html">java.lang.Deprecated</a>
+<p>
+Experimental: use <a href="https://developer.android.com/reference/androidx/annotation/RequiresOptIn">androidx.annotation.RequiresOptIn</a>
+   </td>
+   <td>/**
+<p>
+@deprecated As of release x.x, replaced by {@link #newMethod()}
+<p>
+*/
+<p>
+/**
+<p>
+Warning: This API is experimental
+<p>
+*/
+   </td>
+  </tr>
+  <tr>
+   <td>Apple
+   </td>
+   <td>In Objective-C:
+<p>
+<code>__attribute__((deprecated("Use newMethod instead")));</code>
+<p>
+In Swift:
+<p>
+<code>@available(*, deprecated, message: "Use newMethod instead")</code>
+   </td>
+   <td>In Objective-C:
+<p>
+/**
+<p>
+* @deprecated Use `newMethod` instead.
+<p>
+*/
```

Comment:
Would be nice to make these code blocks, though I don't know if that works in a table.
```suggestion
/**
<pre><code>
<p>
* @deprecated Use `newMethod` instead.
<p>
</code></pre>
*/
```

Or maybe triple-backquote will work here.

Same for the swift example